src/components/grid-list/grid-list.ts

```diff
@@ -1,8 +1,127 @@
   templateUrl: './components/grid-list/grid-list.html',
   styleUrls: ['./components/grid-list/grid-list.css'],
+  encapsulation: ViewEncapsulation.None
 })
-export class MdGridList {}
+export class MdGridList implements OnInit, AfterContentChecked {
+  /** Number of columns being rendered. */
+  _cols: number;
+
+  /** Row height value passed in by user */
+  _rowHeight: string;
+
+  /** The amount of space between tiles. This will be something like '5px' or '2em'. */
+  _gutter: string = '1px';
+
+  /** Sets position and size styles for a tile */
+  _tileStyler: TileStyler;
+
+  /* Array of grid tiles being rendered (from query list) */
+  _tiles: MdGridTile[] = [];
+
+  constructor(private _renderer: Renderer, private _element: ElementRef,
+              private _dir: Dir) {}
+
+  get cols() { return this._cols; }
+
+  @Input('cols')
+  set cols(value: any) { this._cols = ensureNumberInput(value); }
+
+  get gutterSize() { return this._gutter; }
+
+  @Input('gutterSize')
+  set gutterSize(value: any) {
+    this._gutter = ensureStringInput(value);
+  }
+
+  /** Set internal representation of row height from the user-provided value. */
+  @Input('rowHeight')
+  set rowHeight(value: string | number) {
+    this._rowHeight = ensureStringInput(value);
+    if (value === RowHeightMode.FIT) {
+      this._tileStyler = new FitTileStyler();
+    } else if (this._rowHeight && this._rowHeight.match(/:/g)) {
+      this._tileStyler = new RatioTileStyler(this._rowHeight);
+    } else {
+      this._tileStyler = new FixedTileStyler(this._rowHeight);
+    }
+  }
+
+  /** Query list of tiles that are being rendered. */
+  @ContentChildren(MdGridTile)
+  set tiles(tiles: QueryList<MdGridTile>) {
+    this._tiles = tiles.toArray();
+  }
+
+  ngOnInit() { this._checkDefaults(); }
+
+  ngAfterContentChecked() { this.layoutTiles(); }
+
+  _checkDefaults(): void {
+    // if cols property is missing, throw a friendly error
+    if (!this.cols) {
+      throw `md-grid-list: must pass in number of columns. Example: <md-grid-list cols="3">`;
+    }
+    // if rowHeight property is missing, default to equal width:height
+    if (!this._rowHeight) {
+      this._tileStyler = new RatioTileStyler('1:1');
+    }
+  }
+
+  /** Computes and applies the size and position for all children grid tiles. */
+  layoutTiles(): void {
+    let tracker = new TileCoordinator(this.cols, this._tiles);
+    this._tileStyler.init(this.gutterSize, tracker, this.cols, this._dir);
+
+    for (let i = 0; i < this._tiles.length; i++) {
+      let pos = tracker.positions[i];
+      let tile = this._tiles[i];
+      this._tileStyler.setStyle(tile, pos.row, pos.col);
+    }
+    this.setListStyle(this._tileStyler.getComputedHeight());
+  }
+
+  setListStyle(style: string[]): void {
+    if (style) {
+      this._renderer.setElementStyle(this._element.nativeElement, style[0], style[1]);
+    }
+  }
+}
+
+export function ensureStringInput(value: string | number): string {
```

Comment:
The name of this function should be something like `coerceToString` or `makeString` or something like that. `ensure` sounds like it is doing some assertion.